File: 5_Go/algorithm/sort/insertionsort.go
package sort

func InsertionSort(arr []int) {
	size := len(arr)
	for i := 1; i < size; i++ {
		for j = i; j > 0; j-- {
			if arr[j] < arr[j-1] {
				arr[j], arr[j-1] = arr[j-1], arr[j]
			}
		}
	}
}

File: 2_Python/Algorithm/Sorting/insertionsort.py
def insertion_sort(array):
    """
    Insertion sort implementation

    :param array: unsorted array
    :return: None
    """
    size = len(array)
    for i in range(1, size):
        for j in range(i, 0, -1):
            if array[j] < array[j-1]:
                array[j], array[j-1] = array[j-1], array[j]

